ddtrace/opentelemetry/tracer_test.go

```diff
@@ -42,7 +46,7 @@ func TestSpanWithContext(t *testing.T) {
 
 	assert.True(ok)
 	assert.Equal(got, sp.(*span).Span)
-	assert.Equal(fmt.Sprintf("%x", got.Context().SpanID()), sp.SpanContext().SpanID().String())
+	assert.Equal(fmt.Sprintf("%x", got.Context().SpanID()), strings.TrimLeft(sp.SpanContext().SpanID().String(), "0"))
```

Comment:
Are we trimming the zeroes off the span id?

Reply:
I think `sp.SpanContext().SpanID().String()` preserves leading zeroes of the hex encoding of the Span ID so that the length is always 16, but `fmt.Sprintf("%x", got.Context().SpanID())` does not, and we were running into some trouble with this test being flaky because of this. But I think 

`	
assert.Equal(fmt.Sprintf("%016x", got.Context().SpanID()), sp.SpanContext().SpanID().String())
`
Is a better way to fix the flaky behavior, so I'll update it to this!